### PR TITLE
Upgrade to react-native-windows 0.72.2

### PR DESCRIPTION
**Pull request by JunielKatarn**
## Description
Upgrade React Native Windows to version 0.72.2

### Why
CxxModule types are discouraged and will eventually be deprecated.

### What
Upgrades the networking stack to the modern TurboModule types.

### CODE REVIEW

File: windows/rngallery/rngallery.vcxproj.filters [hunk 1]
```diff
@@ -41,141 +41,6 @@
     <Image Include="Assets\LockScreenLogo.scale-200.png">
       <Filter>Assets</Filter>
     </Image>
-    <Image Include="Assets\StoreLogo.scale-200.png">
```
Comment (jonthysell) [inline windows/rngallery/rngallery.vcxproj.filters:44]: Plz revert. We still want our nice icons. ;)

File: windows/rngallery/rngallery.vcxproj [hunk 2]
```diff
@@ -114,58 +114,13 @@
     </AppxManifest>
   </ItemGroup>
   <ItemGroup>
-    <Image Include="Assets\LargeTile.scale-100.png" />
```
Comment (jonthysell) [inline windows/rngallery/rngallery.vcxproj:117]: Plz revert. We still want our nice icons. ;)

File: windows/rngallery/Package.appxmanifest [hunk 3]
```diff
@@ -4,23 +4,18 @@
   xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
```
Comment (jonthysell) [inline windows/rngallery/Package.appxmanifest:4]: Plz revert the changes to this file.

File: windows/rngallery/MainPage.cpp [hunk 4]
```diff
@@ -17,19 +16,5 @@ namespace winrt::rngallery::implementation
         InitializeComponent();
         auto app = Application::Current().as<App>();
         ReactRootView().ReactNativeHost(app->Host());
-        // Workaround for https://github.com/microsoft/react-native-windows/issues/6287
```
Comment (jonthysell) [inline windows/rngallery/MainPage.cpp:20]: Revert? This looks like this should still be here.

File: windows/rngallery/Assets/SplashScreen.scale-200.png
Comment (jonthysell) [inline windows/rngallery/Assets/SplashScreen.scale-200.png:1]: Please revert all the icon changes, I still think we need our icons.

**Review by jonthysell — CHANGES_REQUESTED** [review]
I think the only changes that need to happen here are to `package.json`.

File: windows/rngallery/MainPage.cpp (hunk 4 above)
Comment (JunielKatarn) [reply windows/rngallery/MainPage.cpp:20]: Sure. Note, I only ran the `react-init --override` command. No manual changes so far.

**Comment by JunielKatarn** [comment]
> I think the only changes that need to happen here are to `package.json`.

I agree, but the react-init command did the rest.
I have reverted everything that doesn't make sense.
@jonthysell can you review once more?

File: windows/rngallery/MainPage.xaml [hunk 5]
```diff
@@ -8,12 +8,9 @@
...
     Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
-    <Grid>
-        <Canvas x:Name="RNPrintCanvas" Opacity="0" />
-        <react:ReactRootView 
-            x:Name="ReactRootView"
-            ComponentName="rngallery"
-            Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
-            MinHeight="400"/>
-    </Grid>
+    <react:ReactRootView
```
Comment (JunielKatarn) [inline windows/rngallery/MainPage.xaml:11]: Another automatic change.
@jonthysell  Should I revert?

The app loads fine with these changes.

**Comment by jonthysell** [comment]
Gallery is already on 73.
